### DISCUSSION TITLE
Support `discriminator` for response object selection

### DISCUSSION THREAD

**Discussion by equalspeterd** [stoplightio/prism#523] [Ideas]
**User story.**
As an API designer, I want to define more generalized APIs that have nearly identical semantics, but have different response objects based on a path parameter, detailed by the `discriminator` property.

**Is your feature request related to a problem?**
The current mock server randomly chooses a schema (x or y), regardless of the value of {dataset} (see example below). Also, currently, when a full example is defined for one of the response types, only that response is used in mock responses. I would have expected continued random selection between the 2 example schemas.

**Describe the solution you'd like**
Support for `discriminator` in general.

**Additional context**
e.g.

```
  /info/{dataset}:
    parameters:
      - name: dataset
        required: true
        in: path
        schema:
          type: string
          enum:
            - x
            - y
      get:
          content:
            application/json:
              schema:
                oneOf:
                  - $ref: '#/components/schemas/xList'
                  - $ref: '#/components/schemas/yList'
                discriminator:
                  propertyName: dataset
                  mapping:
                    x: '#/components/schemas/xList'
                    y: '#/components/schemas/yList'

```

**Comment by philsturgeon** [comment]
Support for the discriminator keyword needs to happen in two places:

1. Input/Output Validation (and contract testing)
2. Mock Generation (somewhere above JSON Schema Faker, either http-spec or prism somewhere)

Implementing in 1 is possible, 2... is tricky. One idea would be trying to convert discriminator to if/then/else in https://github.com/mikunn/openapi-schema-to-json-schema/

**Comment by philsturgeon** [comment]
This is on hold until we figure out if discriminator is going to be deprecated or not. https://github.com/OAI/OpenAPI-Specification/issues/2143

**Reply by Gvade** [reply]
Are there any updates?

**Comment by MihaelMihov** [comment]
Any updates here guys? The dsiscriminator is not going to be deprecated in OpenAPI spec 3.0 any time near soon.
